### ISSUE TITLE
Extend the allowed characters on the `validUrlQueryEndingChars`

### ISSUE THREAD

**Issue by FMRb** [twitter/twitter-text#166]
The valid characters allow on the `validUrlQueryEndingChars` is limited and it could be extended based on the [RFC3986](https://tools.ietf.org/html/rfc3986) section 2.2, supporting characters like `)` closing parentheses.
```js
twttr.txt.regexen.validUrlQueryEndingChars = /[a-z0-9_&=#\/]/i
```
https://github.com/twitter/twitter-text/blob/master/js/pkg/twitter-text-1.14.3.js#L267

**Comment by FMRb** [comment]
@codemonkey3045 any update on the issue?

**Comment by codemonkey3045** [comment]
Sorry, I've gotten busy with a few other things. For this one, we would have to refactor the code a bit so it'll take a little longer to factor into my work schedule. I'm concerned about cases where someone will tweet 

> Hey guys, check out my new lib (https://github.com/whatever)

That last ) isn't intended to be part of the url. If we linkify it together, that'll be a broken link. For every preceding (, we need to trim a ) from the end.

**Comment by FMRb** [comment]
Thanks for taking the time, it makes completely sense what you described. 
I'll try to find some time to work on this case.

**Comment by CoryDanielson** [comment]
Could we get some of these characters supported in the meantime, but leave `)` until later? I'm experiencing this same issue with a URL query ending in `-`

- https://github.com/twitter/twitter-text/issues/190

**Comment by kaushlakers** [comment]
Hey this has been fixed with the 2.0 update. Feel free to reopen if you find otherwise.
